src/main/kotlin/app/revanced/utils/adb/Commands.kt

```diff
@@ -17,7 +18,19 @@ internal fun JadbDevice.buildCommand(command: String, su: Boolean = true): Shell
 }
 
 internal fun JadbDevice.run(command: String, su: Boolean = true): Int {
-    return this.buildCommand(command, su).start().waitFor()
+    if (su) {
+        return this.buildCommand(command, su).start().waitFor()
+    }
+
+    // Avoid deadlock whilst checking for root access
+    val adbCommand = this.buildCommand(command, su).start()
+    val byteArray = ByteArray(512)
+    val inputStream = adbCommand.inputStream
+    while (inputStream.read(byteArray) != -1){
+        Constants.SUPERSU = true
+    }
+    inputStream.close()
```

Comment:
Maybe this should be done differently? It doesn't look very good.

Reply:
Updated, uncertain if I shall extract it to a different function(?)